apps/dot_ext/management/commands/authflow_id_delete_and_vacuume.py

```diff
@@ -0,0 +1,67 @@
+import time
+from django.db import connection
+
+from datetime import date, timedelta
+from django.core.management.base import BaseCommand
+
+from apps.dot_ext.models import AuthFlowUuidCopy
+
+DELETE_WITH_LIMIT = "DELETE FROM {table_name} WHERE auth_uuid IN (SELECT auth_uuid FROM {table_name} WHERE created < '{age_date}'::date ORDER BY created LIMIT {limit_on_delete})"
+VACUUM_TABLE = "VACUUM FULL {table_name}"
+
+
+def delete_and_vacuume_auth_uuid_table(age, limit_on_delete, run_vacuum, dry_run):
+    '''
+    Delete aged authflow uuid records, and then
+    call VACUUM FULL (Postgres only) on table: dot_ext_authflowuuid
+    '''
+
+    days_before = date.today()-timedelta(days=age)
+    # change table name dot_ext_authflowuuidcopy to dot_ext_authflowuuid before merge
+    delete_sql = DELETE_WITH_LIMIT.format(table_name="dot_ext_authflowuuidcopy", age_date=days_before.strftime("%m/%d/%Y"), limit_on_delete=limit_on_delete)
+    vacuum_sql = VACUUM_TABLE.format(table_name="dot_ext_authflowuuidcopy")
+
+    start_t = time.time()
+
+    aged_records_total = AuthFlowUuidCopy.objects.filter(created__lt=days_before).count()
+
+    print("Aged records total = {}".format(aged_records_total))
+
+    with connection.cursor() as cursor:
+
+        print("EXECUTE SQL : {}".format(delete_sql))
+
+        if not dry_run:
+            cursor.execute(delete_sql)
+
+        end_t = time.time()
+
+        print("Delete aged records: time elapsed (sec) = {}".format(end_t - start_t))
+
+        if run_vacuum:
+
+            print("EXECUTE SQL : {}".format(vacuum_sql))
+
+            if not dry_run:

```

Comment:
Just curious... what does doing the dry run actually do? Might be useful if it as least output the full query plan?

Reply:
print out the actual sequence of SQL statements would be executed.
since full table vacuum is optional, so a dry run shows what statements would be executed.

Reply:
as shown in the sample run (PR template section: What Should Reviewers Watch For?), the time elapsed are as show:

**---------------------------- RUN 1 (VACCUM backlogged dead records)---------------------------------
python manage.py authflow_id_delete_and_vacuume -a 30 -l 30000 -u
delete_sql : DELETE FROM dot_ext_authflowuuidcopy WHERE auth_uuid IN (SELECT auth_uuid FROM dot_ext_
authflowuuidcopy WHERE created < '08/04/2021'::date ORDER BY created LIMIT 30000)
vacuum_sql : VACUUM FULL dot_ext_authflowuuidcopy
aged records total = 449458
query + delete aged records: time elapsed (sec) = 0.2269577980041504
query + delete aged records and vacuum table: time elapsed (sec) = 1.1252453327178955
------------------------------------- RUN 2 ---------------------------------------------------------
python manage.py authflow_id_delete_and_vacuume -a 30 -l 30000 -u
delete_sql : DELETE FROM dot_ext_authflowuuidcopy WHERE auth_uuid IN (SELECT auth_uuid FROM dot_ext_
authflowuuidcopy WHERE created < '08/04/2021'::date ORDER BY created LIMIT 30000)
vacuum_sql : VACUUM FULL dot_ext_authflowuuidcopy
aged records total = 419458
query + delete aged records: time elapsed (sec) = 0.21963214874267578
query + delete aged records and vacuum table: time elapsed (sec) = 0.9247055053710938
-------------------------------------- RUN 3 --------------------------------------------------------
....
**

Reply:
these are very short tx, showing exec plan would be value for some BFD query analysis where slowness and degradation of performance need to be addressed...